lib/solidus_subscriptions/processor.rb

```diff
@@ -4,32 +4,17 @@ module SolidusSubscriptions
   class Processor
     class << self
       def run
-        SolidusSubscriptions::Subscription.actionable.find_each(&method(:process_subscription))
-        SolidusSubscriptions::Installment.actionable.find_each(&method(:process_installment))
+        SolidusSubscriptions::Subscription
+          .where(installments: SolidusSubscriptions::Installment.actionable)
+          .or(SolidusSubscriptions::Subscription.actionable)
+          .distinct
+          .find_each(&method(:process_subscription))
       end
 
       private
 
       def process_subscription(subscription)
```

Comment:
You can remove this method and inline it in `#run`, now that we're only processing subscriptions.

Reply:
Also this makes sense, it's more explicit.